File: prime/src/main/java/com/telenordigital/prime/disruptor/ClearingEventHandler.java
package com.telenordigital.prime.disruptor;

import com.lmax.disruptor.EventHandler;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class ClearingEventHandler implements EventHandler<PrimeEvent> {

    private static final Logger LOG = LoggerFactory.getLogger(ClearingEventHandler.class);

    @Override
    public void onEvent(
            final PrimeEvent event,
            final long sequence,
            final boolean endOfBatch) {
        try {
            event.clear();
        } catch (Exception e) {
            LOG.warn("Exception clearing the prime event", e);
        }
    }
}

File: ocs/src/main/kotlin/org/ostelco/prime/disruptor/ClearingEventHandler.kt
package org.ostelco.prime.disruptor

import com.lmax.disruptor.EventHandler
import org.ostelco.prime.logger

class ClearingEventHandler : EventHandler<OcsEvent> {

    private val logger by logger()

    override fun onEvent(
            event: OcsEvent,
            sequence: Long,
            endOfBatch: Boolean) {
        try {
            event.clear()
        } catch (e: Exception) {
            logger.warn("Exception clearing the prime event", e)
        }
    }
}
